cdap-securestore-ext-cloudkms/src/main/java/co/cask/cdap/securestore/gcp/cloudkms/CloudKMSClient.java

```diff
@@ -0,0 +1,246 @@
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.securestore.gcp.cloudkms;
+
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
+import com.google.api.client.googleapis.json.GoogleJsonResponseException;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.cloudkms.v1.CloudKMS;
+import com.google.api.services.cloudkms.v1.CloudKMSScopes;
+import com.google.api.services.cloudkms.v1.model.CryptoKey;
+import com.google.api.services.cloudkms.v1.model.DecryptRequest;
+import com.google.api.services.cloudkms.v1.model.DecryptResponse;
+import com.google.api.services.cloudkms.v1.model.EncryptRequest;
+import com.google.api.services.cloudkms.v1.model.EncryptResponse;
+import com.google.api.services.cloudkms.v1.model.KeyRing;
+import com.google.common.io.CharStreams;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.ByteArrayInputStream;
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import javax.annotation.Nullable;
+
+/**
+ * Wrapper on {@link CloudKMS} client.
+ */
+class CloudKMSClient implements Closeable {
+  private static final Logger LOG = LoggerFactory.getLogger(CloudKMSClient.class);
+  // When created in the global location, Cloud KMS resources are available from zones spread around the world.
+  private static final String LOCATION_ID = "global";
+  // All the crypto keys are created under keyring named cdap
+  private static final String KEYRING_ID = "cdap";
+  private static final String ENCRYPT_DECRYPT = "ENCRYPT_DECRYPT";
+  private static final String CLOUD_KMS = "cloudkms";
+  private static final String PROJECT_ID = "projectId";
+  private static final String ACCOUNT_KEY = "accountKey";
+  private static final String METADATA_SERVER_API = "metadataApi";
+
+  private final CloudKMS cloudKMS;
+  private final String projectId;
+  private final String accountKey;
+  private final String metadataServerApi;
+  // In-memory cache to hold created crypto keys, this is to avoid checking if a given crypto key exists.
+  private final Set<String> crypoKeyList;
```

Comment:
Shouldn't name a `Set` with `List` in the name. It is confusing. Just call it `knownCryptoKeys`